features/util.py

```diff
@@ -23,7 +23,10 @@ def launch_lxd_container(
     subprocess.run(["lxc", "launch", image_name, container_name])
 
     def cleanup_container() -> None:
-        subprocess.run(["lxc", "delete", "-f", container_name])
+        if context.config.preserve_instance:
+            print("Leaving lxc container running: {}".format(container_name))
```

Comment:
Minor one, but the step definitions refer to them as "lxd containers", and we can use `print`s behaviour to simplify a little:

```suggestion
            print("Leaving lxd container running:", container_name)
```